Sources/Apollo/GraphQLResultNormalizer.swift

```diff
@@ -3,47 +3,76 @@ import Foundation
 import ApolloAPI
 #endif
 
-final class GraphQLResultNormalizer: GraphQLResultAccumulator {
+struct ResultNormalizerFactory {
+  private init() {}
+
+  static func selectionSetDataNormalizer() -> SelectionSetDataResultNormalizer {
```

Comment:
I decided to create separate functions rather than `normalizer(for: DataSource)` because this way, each function returns the concrete type (instead of `BaseGraphQLResultNormalizer`). This allows the generic functions in the `GraphQLExecutor` to still specialize the function calls, using static dispatch instead of dynamic dispatch.